config/environment.rb

```diff
@@ -1,5 +1,8 @@
 # Load the Rails application.
 require File.expand_path("../application", __FILE__)
 
+# Load custom exceptions
+Dir[Rails.root.join("lib/exceptions/**/*.rb")].each { |f| require f }
```

Comment:
Why not put the exceptions in the models directory? If you have them in the models directory, they will benefit from Rails auto loading/reloading features for development and test.

Reply:
I suppose the rake task you created wouldn't be able to find them anymore, but you could tweak the task to eager load classes.